plugins/user-authenticators/oauth2/src/main/java/org/apache/cloudstack/oauth2/OAuth2Utils.java

```diff
@@ -0,0 +1,28 @@
+package org.apache.cloudstack.oauth2;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeFlow;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.store.MemoryDataStoreFactory;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+public class OAuth2Utils {
```

Comment:
This class looks very specific to Google, we can rather rename it or let the Google auth be the implementation of an abstract method in the oauth framework

Reply:
Yes we can do that. AuthorizationCodeFlow class can be used to connect to any oauth server. So it makes sense to keep only the generic methods in the util class. But as of now, I'm keen towards building a minimum viable product with only google as idp. Once it is done, I'll tidy up the Util class to only include methods that can be used by any oauth server.